LuaFunctions.cpp

```diff
@@ -809,6 +809,7 @@ ElunaRegister<Creature> CreatureMethods[] =
     { "SetEquipmentSlots", &LuaCreature::SetEquipmentSlots },
 
     // Boolean
+    { "IsDungeonBoss", &LuaCreature::IsWorldBoss },
```

Comment:
This requires to be surrounded with ifdef as well. Also, looks like there is a copy-paste mistake with the `LuaCreature::IsWorldBoss`.
```suggestion
#if defined(TRINITY) || defined(AZEROTHCORE)
    { "IsDungeonBoss", &LuaCreature::IsDungeonBoss},
#endif
```